acme/acme/client.py

```diff
@@ -255,16 +258,23 @@ def retry_after(cls, response, default):
         :rtype: `datetime.datetime`
 
         """
-        retry_after = response.headers.get('Retry-After', str(default))
+        retry_after = response.headers.get('Retry-After', str(default)).strip()
         try:
             seconds = int(retry_after)
         except ValueError:
             # pylint: disable=no-member
-            decoded = werkzeug.parse_date(retry_after)  # RFC1123
-            if decoded is None:
+            when = parsedate_tz(retry_after)
+            try:
+                year = when[0] # raises TypeError if t is None
+                # py26: Handle two-digit years -- but any server that thinks
+                # "retry after 99" means "come back after 1999" is.. deprecated
+                if year >= 0 and year < 100:
+                    when = [year + 2000] + list(when[1:]) #  pragma: no cover
+                tzone = when[-1] if when[-1] else 0
+                # raises ValueError/OverflowError
```

Comment:
try/except block should be limited to the smallest possible block of code, in this case line below